block-storage/src/main/scala/coop/rchain/blockstorage/util/io/errors.scala

```diff
@@ -0,0 +1,75 @@
+package coop.rchain.blockstorage.util.io
+
+import java.io.{FileNotFoundException, IOException}
+import java.nio.file.{
+  AtomicMoveNotSupportedException,
+  DirectoryNotEmptyException,
+  FileAlreadyExistsException,
+  NotDirectoryException
+}
+
+import cats.data.EitherT
+
+sealed trait IOError
+
+final case class FileSeekFailed(exception: IOException)                     extends IOError
+final case class IntReadFailed(exception: IOException)                      extends IOError
+final case class ByteArrayReadFailed(exception: IOException)                extends IOError
+final case class IntWriteFailed(exception: IOException)                     extends IOError
+final case class ByteArrayWriteFailed(exception: IOException)               extends IOError
+final case class SetLengthFailed(exception: IOException)                    extends IOError
+final case class ClosingFailed(exception: IOException)                      extends IOError
+final case class FileNotFound(exception: FileNotFoundException)             extends IOError
+final case class FileSecurityViolation(exception: SecurityException)        extends IOError
+final case class FileIsNotDirectory(exception: NotDirectoryException)       extends IOError
+final case class UnsupportedFileOperation(e: UnsupportedOperationException) extends IOError
+final case class FileAlreadyExists(e: FileAlreadyExistsException)           extends IOError
+final case class DirectoryNotEmpty(e: DirectoryNotEmptyException)           extends IOError
+final case class AtomicMoveNotSupported(e: AtomicMoveNotSupportedException) extends IOError
+final case class UnexpectedIOError(throwable: Throwable)                    extends IOError
+
+object IOError {
+  type IOErr[A]        = Either[IOError, A]
```

Comment:
I find this name misleading. You have code like ```someFunction(a) : F[IOErr[String]]``` and you would read this as "somefunction returns an error wrapped in F"

Maybe `IO` or `IOAction` would be a better name